### PR TITLE
xkbcommon: Bump wayland-protocols and required_conan_version

### PR DESCRIPTION
**Pull request by jwillikers**
Specify library name and version:  **xkbcommon/***

Fix building in the build context.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/xkbcommon/all/conanfile.py [hunk 1]
```diff
@@ -39,7 +40,7 @@ class XkbcommonConan(ConanFile):
 
     @property
     def _has_build_profile(self):
-        return hasattr(self, "settings_build")
+        return hasattr(self, "settings_build") and cross_building(self)
```
Comment (SpaceIm) [inline recipes/xkbcommon/all/conanfile.py:43]: No change is required here I would say.

Comment (jwillikers) [reply recipes/xkbcommon/all/conanfile.py]: You're right, that would change the method meaning. I was trying to fix the issue that you also ran into in #20542. I'm not sure id didn't cause other problems. I've reverted.

**Review by danimtb — DISMISSED** [review]

File: recipes/xkbcommon/all/conanfile.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 from conan import ConanFile
 from conan.tools.apple import fix_apple_shared_install_name
+from conan.tools.build import cross_building
```
Comment (ericLemanissier) [inline recipes/xkbcommon/all/conanfile.py:5]:
```suggestion
```

**Review by ericLemanissier — CHANGES_REQUESTED** [review]

**Review by ericLemanissier — APPROVED** [review]

File: recipes/xkbcommon/all/conanfile.py [hunk 3]
```diff
@@ -82,7 +82,7 @@ def build_requirements(self):
             self.tool_requires("pkgconf/2.0.3")
         if self._has_build_profile and self.options.get_safe("with_wayland"):
             self.tool_requires("wayland/<host_version>")
-            self.tool_requires("wayland-protocols/1.31")
+            self.tool_requires("wayland-protocols/1.32")
```
Comment (uilianries) [inline recipes/xkbcommon/all/conanfile.py:85]:
```suggestion
            self.tool_requires("wayland-protocols/<host_version>")
```

https://docs.conan.io/2/reference/conanfile/methods/build_requirements.html#host-version

File: recipes/xkbcommon/all/conanfile.py [hunk 4]
```diff
@@ -82,7 +82,7 @@ def build_requirements(self):
             self.tool_requires("pkgconf/2.0.3")
         if self._has_build_profile and self.options.get_safe("with_wayland"):
             self.tool_requires("wayland/<host_version>")
-            self.tool_requires("wayland-protocols/1.31")
+            self.tool_requires("wayland-protocols/<host_version>")
```
Comment (ericLemanissier) [inline recipes/xkbcommon/all/conanfile.py:85]:
```suggestion
            self.tool_requires("wayland-protocols/1.32")
```
host_version cannot work when the dependency is either a build requirement or a requirement, but not both.

**Review by ericLemanissier — CHANGES_REQUESTED** [review]

**Review by ericLemanissier — APPROVED** [review]

**Comment by SpaceIm** [comment]
wayland-protocols should be removed from requirements, and another workaround found for 1 profile.

**Comment by jwillikers** [comment]
> wayland-protocols should be removed from requirements, and another workaround found for 1 profile. It's bad to have requirements depending on 1 or 2 profiles, because it corrupts dependency graph. Requirements (not build requirements) must never depend on settings_build.

Would it be possible to just require two profiles using the `validate` method? That would make things simpler.

**Comment by SpaceIm** [comment]
I've submitted https://github.com/conan-io/conan-center-index/pull/21239. I think it should work but I can't test it right now.

**Comment by jwillikers** [comment]
> I've submitted #21239. I think it should work but I can't test it right now.

Thanks! I'll double check that yours cross-compiles correctly tomorrow.

**Comment by jwillikers** [comment]
Closing in favor of #21239

**Pull request closed without merging**